### PR TITLE
bpo-45847: Port _multiprocessing to PY_STDLIB_MOD (GH-29768)

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue45847

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -6253,6 +6258,10 @@ PY_STDLIB_MOD_SIMPLE([_typing])
 PY_STDLIB_MOD_SIMPLE([_xxsubinterpreters])
 PY_STDLIB_MOD_SIMPLE([_zoneinfo])
 
+dnl multiprocessing modules
+PY_STDLIB_MOD_SIMPLE([_posixsubprocess])
+PY_STDLIB_MOD_SIMPLE([_multiprocessing], [-I'$(srcdir)/Modules/_multiprocessing'])
```
Comment (tiran) [inline configure.ac]: This does not work, you have to use backslash quoting
```
MODULE__MULTIPROCESSING_CFLAGS=-I'/Modules/_multiprocessing'
```

**Comment by erlend-aasland** [comment]
AFAICS, `HAVE_SEM_OPEN` and `POSIX_SEMAPHORES_NOT_ENABLED` are handled by `#ifdef`s in the source code. We _could_ add a check for `$ac_cv_header_fcntl_h`, but it's probably not needed.

**Review by tiran — APPROVED** [review]
Thanks, looks good to me.

I'm unsure about the semaphore extension. It is hard to test.

**Comment by erlend-aasland** [comment]
> I'm unsure about the semaphore extension.

Me too. Let's ask on python-dev / Discourse.

**Review by tiran — APPROVED** [review]

**Pull request merged**